main.go

```diff
@@ -615,4 +675,18 @@ func getRouter(logger *zap.Logger, config *Config, indexer Indexer, cache *expir
 
 // healthHandler is used for Docker/K8s deployments. It returns 200 if the service is live
 // In addition ?ready=true can be used for a ready request. Currently both are identical.
-func healthHandler(w http.ResponseWriter, r *http.Request) {}
+func healthHandler(options handlerOptions) func(http.ResponseWriter, *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		for k := range r.URL.Query() {
+			switch k {
+			case "ready":
+				// Ready check, currently same as live check
+			default:
+				if !options.allowUnknownQueryParameters {
+					badRequest(w, fmt.Sprintf("unknown query parameter: %s", k))
+				}
+				return
+			}
+		}
+	}
+}
```

Comment:
For this handler, I've followed the description in the comment of this function. There it is stated that it can be used `?ready=true` even if there is no different logic currently for that.
Therefore, this endpoint just allows one query parameter (`ready`), any other parameter is going to return a 400 error code.